proxy/apiserver/router/router.go

```diff
@@ -111,6 +120,74 @@ func (r *router) Route(httpReq *http.Request, reqInfo *request.RequestInfo) (*Ro
 	return &RouteAccept{}, nil
 }
 
+func (r *router) injectSelector(httpReq *http.Request, gvr *schema.GroupVersionResource, snapshot *proxyclient.ProtoSpecSnapshot) error {
+	protoSpec, _, err := snapshot.AcquireSpec()
```

Comment:
We don't have to acquire spec here, it can be a parameter of this function and reuse `protoSpec` in `Route` function.